Added an fps limiter toggle in game, graphical settings automation and controller auto enable.

TroyWarez opened the pull request:
I ran bully vanilla with only the updated silent patch and I couldn't get it to crash even after 6+ hours of gameplay on Windows 11 24H2. I believe the updated toolset has under the hood improvement with possible smart pointers to add a whole new level of stability. The game seems less stable at very high resolutions (4K+), I don't think it's a regression but, rather a flaw of the old engine. *Incorrect information

I also removed the mini dumper because it would cause Steam to lose focus when in Big Picture mode. It also can take up a lot of disk space with little benefit to most users. Feel free to add it back or make any other required changes. To test this, try to change the resolution of the Bully after loading into the game first or any other way of crashing the game. Steam should not respond to any controller input when in big picture mode. *Edit: More context added, testers need to confirm that the mini dumper is the cause.

Here's a release with some updated loader files and a new .ini here: https://github.com/TroyWarez/SilentPatchBully/releases/tag/BUILD-4-(BETA)
Release notes:  *Edit: Updated release notes
- Added an in-game fps toggle button combination (BACK/SELECT + DPAD DOWN) or F11 with your keyboard
- Added an automatic settings changer to the .ini file, see the comments inside the default config for more details
- Added an automatic file changer to enable controllers by default or disable them. (Make sure to have your controller connected before starting the game)

TroyWarez committed:
first commit

TroyWarez committed:
vs

TroyWarez committed:
vs 2017

TroyWarez committed:
in game fps toggle

TroyWarez committed:
removed debug loop

TroyWarez committed:
added auto xinput toggle in config

TroyWarez committed:
Added fps toggle with keyboard

TroyWarez committed:
fixed config

TroyWarez committed:
fixed file handling

TroyWarez committed:
removed path

TroyWarez committed:
Update README.md

CookiePLMonster commented:
I will review this properly later as there's a lot to unpack, but I'd like to highlight this right away:

> I believe the updated toolset has under the hood improvement with possible smart pointers to add a whole new level of stability.

This is simply impossible and this reasoning makes no sense. There are no known stability issues introduced by the patch itself, all known stability problems are caused by yet unfixed issues in the game. An upgrade of a toolset of the patch virtually cannot influence that, so you are certainly seeing placebo results.

On top of that, I have a policy not to deprecate OS support in patch updates. SP for Bully was started with WinXP support, and I intend to preserve that.

TroyWarez commented:
> This is simply impossible and this reasoning makes no sense. There are no known stability issues introduced by the patch itself, all known stability problems are caused by yet unfixed issues in the game. An upgrade of a toolset of the patch virtually cannot influence that, so you are certainly seeing placebo results.
> 
> On top of that, I have a policy not to deprecate OS support in patch updates. SP for Bully was started with WinXP support, and I intend to preserve that.

Thank you for your insight, I am always happy to learn new information.  I think you're right about it being a placebo. I upgraded the toolset to add support for `RegSetKeyValue` which can be easily replaced with other similar calls.*

*Edit: A test compile was successful on v141_xp with `#include <debugapi.h>` removed

*Edit 2: Desired revision(s):  https://github.com/CookiePLMonster/SilentPatchBully/pull/124/commits/85206a727e20f6399e6c08c11a635bfb4fcd5835

I reverted the platform toolset to 141_xp and solution to vs15 and fixed a bug with the key being held. The keyboard toggle also works when a controller is disconnected now. 

The implementation of QOL features was the main goal of the fork, the platform toolset upgrade was a side effect and I updated the title to reflect that.

*Edit 3: I updated the release files with the latest build and changed the release notes.

TroyWarez committed:
Reverted to 141_xp, vs 2015 solution file and fixed broken keyboard handling. Replaced RegSetKeyValueW with RegSetKeyValueWLegacy.
